src/steadystateproblem.cpp

```diff
@@ -363,37 +411,40 @@ bool SteadystateProblem::getSensitivityFlag(const Model *model,
 realtype SteadystateProblem::getWrmsNorm(const AmiVector &x,
                                          const AmiVector &xdot,
                                          realtype atol,
-                                         realtype rtol) {
+                                         realtype rtol,
+                                         AmiVector &ewt) {
     N_VAbs(x.getNVector(), ewt.getNVector());
     N_VScale(rtol, ewt.getNVector(), ewt.getNVector());
     N_VAddConst(ewt.getNVector(), atol, ewt.getNVector());
     N_VInv(ewt.getNVector(), ewt.getNVector());
     return N_VWrmsNorm(xdot.getNVector(), ewt.getNVector());
 }
 
-bool SteadystateProblem::checkConvergence(const Solver *solver, Model *model) {
+bool SteadystateProblem::checkConvergence(const Solver *solver,
+                                          Model *model,
+                                          SensitivityMethod checkSensitivities) {
     /* get RHS and compute weighted error norm */
-    model->fxdot(t, x, dx, xdot);
-    wrms = getWrmsNorm(x, xdot, solver->getAbsoluteToleranceSteadyState(),
-                       solver->getRelativeToleranceSteadyState());
+    if (checkSensitivities == SensitivityMethod::adjoint) {
+        getQBfromQ(model, xQ, xQB);
+        getQBfromQ(model, xB, xQBdot);
+        wrms = getWrmsNorm(xQB, xQBdot, solver->getAbsoluteToleranceQuadratures(),
+                           solver->getRelativeToleranceQuadratures(), ewtQB);
+    } else {
+        model->fxdot(t, x, dx, xdot);
+        wrms = getWrmsNorm(x, xdot, solver->getAbsoluteToleranceSteadyState(),
+                           solver->getRelativeToleranceSteadyState(), ewt);
+    }
     bool converged = wrms < RCONST(1.0);
 
-    /* If we also integrate forward sensis, we need to also check those:
-       Check if: sensis enabled && steadyStateSensiMode == simulation */
-    bool checkForwardSensis =
-        solver->getSensitivityOrder() > SensitivityOrder::none &&
-        model->getSteadyStateSensitivityMode() ==
-            SteadyStateSensitivityMode::simulationFSA;
-
     /* get RHS of sensitivities and compute weighted error norm */
-    if (checkForwardSensis) {
+    if (checkSensitivities == SensitivityMethod::forward) {
```

Comment:
Add a comment that `xdot` convergence needs to be checked for both `checkSensitivities == SensitivityMethod::forward` and `SensitivityOrder::none` and why it doesn't need to be checked for `checkSensitivities == SensitivityMethod::adjoint`.

Might be cleaner to have this in a switch statement, but no strong opinion.

Reply:
okay, true. This could be restructured...